### ISSUE TITLE
The examples codes are hard to understand. and missing some explanation.

### ISSUE THREAD

**Issue by wowowooooo** [libp2p/specs#387]
please create a documentation

**Comment by mxinden** [comment]
You are right. Unfortunately we are not doing a great job on documentation. This is mostly due time constraint. We would very much appreciate contributions. In case anyone is up for that, I am happy to answer concrete questions to help folks write such documentation.
